syntax = "proto2";

package crypto.pb;

enum KeyType {
    RSA = 0;
    Ed25519 = 1;
    Secp256k1 = 2;
    ECDSA = 3;
}

message PublicKey {
    KeyType Type = 1;
    bytes Data = 2;
}

message GossipFallbackRequest {
    required PublicKey public_key = 1;
    int32 epoch = 2;
}

message GossipFallbackResponse {
    int32 epoch = 1;
    int32 subnet_id = 2;
    int32 subnet_node_id = 3;
}
